src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -1084,28 +1069,18 @@ ExpressionPtr Visitor::visitTypesafe(Parser::PrimaryExpressionContext* ctx) {
     return visitTypesafe(ctx->brackettedExpression());
   }
 
-  // TODO<joka921> Refactor s.t. try/catch becomes if/else here
   if (ctx->numericLiteral()) {
-    auto literalAny = visitNumericLiteral(ctx->numericLiteral());
-    try {
-      auto intLiteral = literalAny.as<unsigned long long>();
-      return std::make_unique<sparqlExpression::IntExpression>(
-          static_cast<int64_t>(intLiteral));
-    } catch (...) {
-    }
-    try {
-      auto intLiteral = literalAny.as<long long>();
-      return std::make_unique<sparqlExpression::IntExpression>(
-          static_cast<int64_t>(intLiteral));
-    } catch (...) {
-    }
-    try {
-      auto intLiteral = literalAny.as<double>();
-      return std::make_unique<sparqlExpression::DoubleExpression>(
-          static_cast<double>(intLiteral));
-    } catch (...) {
-    }
-    AD_CHECK(false);
+    auto integralWrapper = [](int64_t x) {
+      return ExpressionPtr{
+          std::make_unique<sparqlExpression::IntExpression>(x)};
+    };
+    auto doubleWrapper = [](double x) {
+      return ExpressionPtr{
+          std::make_unique<sparqlExpression::DoubleExpression>(x)};
+    };
+    return std::visit(
+        ad_utility::OverloadCallOperator{integralWrapper, doubleWrapper},
+        visitTypesafe(ctx->numericLiteral()));
```

Comment:
I have thought about further abstractions here, but I am not sure whether they are worth it.
But what you can do, is  (at the beginning of this function)

```
using std::make_unique;
using namespace sparqlExpression;
```
and then write (also above for the strings)
```
      return ExpressionPtr{make_unique<DoubleExpression>(x)};
```
which neatly fits into one line.